### ISSUE TITLE
Azure (Hosted) automerge

### ISSUE THREAD

**Issue by zleight1** [renovatebot/config-help#898]
### Which Renovate are you using?

Renovate Open Source CLI (Hosted)

### Which platform are you using?

Azure DevOps (Hosted)

### Have you checked the logs? Don't forget to include them if relevant

Yes

### What would you like to do?

I'd like to automerge on Azure Repos either with a PR and self approving or with automerge type of branch for certain update types.

I have a repo config like so (the base config is on the hosted k8).

```
{
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "packageRules": [
        {
            "updateTypes": [
                "pin",
                "digest",
                "lockFileMaintenance"
            ],
            "automerge": true
        }
    ],
    "pinDigests": true,
    "pin": {
        "recreateClosed": true,
        "rebaseWhen": "behind-base-branch",
        "stabilityDays": 0,
        "automergeType": "branch"
    },
    "digest": {
        "recreateClosed": true,
        "rebaseWhen": "behind-base-branch",
        "stabilityDays": 0,
        "automergeType": "branch"
    },
    "lockFileMaintenance": {
        "enabled": true,
        "stabilityDays": 0,
        "rebaseWhen": "behind-base-branch",
        "automergeType": "branch"
    },
    "azureAutoComplete": true,
    "semanticCommitScope": "{{parentDir}}"
}
```

I saw that it created the branch and the status was successful (green check mark next to the branch as it ran the `azure-pipeline.yml` and passed).

However, it didn't auto merge despite this. 

My questions:

- Is something wrong here in the config?
- Is this even possible with Azure Repos or is it only GitHub with status checks that would work for this?
- If it's not possible what are my options? We have a branch policy that this would pass, with the exception of a required 1 reviewer. Could RenovateBot bypass and complete itself or self-approve somehow? I see there's a GitHub approver bot, would I have to do this by hand or roll my own approver bot?

**Comment by viceice** [comment]
> Is something wrong here in the config?
Looks good to me. you can the automergeType to the packageRule.

> Is this even possible with Azure Repos or is it only GitHub with status checks that would work for this?
Should work if there are not branch protections blocking the renovate user account.

> If it's not possible what are my options? We have a branch policy that this would pass, with the exception of a required 1 reviewer. Could RenovateBot bypass and complete itself or self-approve somehow? I see there's a GitHub approver bot, would I have to do this by hand or roll my own approver bot?
Renovate can't bypass this, maybe you can exclude the bot user from protection rules?
You can always roll your own bot to auto approve.

**Comment by rarkins** [comment]
What do the debug logs indicate? Most likely it's because if the reviewer requirement but you should verify also that Renovate understands you want to automerge. 

The approver bot is written specifically for GitHub and most platforms don't let you bypass restrictions unless you're an admin, and that's not something I'd recommend.

**Comment by zleight1** [comment]
I’m fairly certain renovate knew I wanted to do a branch strategy as it
didn’t open a PR for the pin but it did for other updates. I’ll clarify
again though.

I will fork the repo and unpin a dep, and rerun the cron job just for the
fork to get an isolated scenario for logs.

When you are talking about branch policies, we do have them set. But
renovateBot is in a group that can bypass these rules either on the PR
checks or push directly to the branch.

Does the fact that the policies exist at all stop renovateBot? Even though
he can bypass them?

Thanks for the rapid responses!

**Comment by viceice** [comment]
We currently don't support pr automerge on azure.

https://github.com/renovatebot/renovate/blob/2a8ad2299044731932c5455d4db02c918a964301/lib/platform/azure/index.ts#L506

So your only way is to use branch merge. For this to work, renovate needs push permissions to the base branch.

So please provide some debug logs, which should show why no automerge happend.

**note:** Renovate needs at least two runs for automerge to work. Can need more runs, if status checks are pending.

**Comment by zleight1** [comment]
@viceice Thanks for the clarification.

So it looks like it thinks the branch is not passing for some reason, despite being OK.

```
Branch status is "yellow" - skipping automerge (repository=Tipalti/FrontEndTooling-renovate-debug, branch=renovate/pin-dependencies)
```
![image](https://user-images.githubusercontent.com/8949977/94802994-42580b00-03f1-11eb-868a-dade1fc18ef5.png)

I took a look at the code that calls this status check and I see how it is automatically green when the checks are disabled, but I am a bit confused as to what it's checking because it looks like it's checking the Behind | Ahead stats as you can see in the picture above.

https://github.com/renovatebot/renovate/blob/5c6442ba973e9a81b06baebc34635f2065a375b1/lib/platform/azure/index.ts#L285

Any ideas? I tried to run the logs as trace but I didn't see anything relevant - I think the pod logs were too long and truncated 🤕

**Comment by viceice** [comment]
As you found, we currently don't support reading status checks from azure, so we only check if our branch is up-to-date.

https://github.com/renovatebot/renovate/issues/1149

**Comment by zleight1** [comment]
> As you found, we currently don't support reading status checks from azure, so we only check if our branch is up-to-date.
> 
> [renovatebot/renovate#1149](https://github.com/renovatebot/renovate/issues/1149)

Mystery solved! I don't think this is too big of a challenge to implement this so I might give it a shot. I already had to write something for a different status check with their API. 

But I still think the check for ahead is wrong, it should be checking you aren't behind as well as the statuses on the branch - no?

**Comment by viceice** [comment]
ok, see https://github.com/renovatebot/renovate/issues/7392

**Comment by viceice** [comment]
> But I still think the check for ahead is wrong, it should be checking you aren't behind as well as the statuses on the branch - no?

Sure, the current check should be removed, as it's not working as intended.

**Comment by zleight1** [comment]
Closing this issue as it was fixed in 23.41.0
